common/web/keyboard-processor/src/keyboards/defaultLayouts.ts

```diff
@@ -535,6 +535,15 @@ export class Layouts {
     return KLS;
   }
 
+  static dfltShiftMultitap: LayoutSubKey = {
+    // Needs to be something special and unique.  Typing restricts us from
+    // using a reserved key-id prefix, though.
+    id: "T_MT_SHIFT_TO_CAPS",
```

Comment:
The typing is there for a reason: it helps us avoid introducing hidden private use types of identifiers in any of the various places these identifiers are used. Leaky abstractions and all that. Yeah, exactly like this case! So this time, at least, the typing is doing its job.

All that said, I'm really happy to look at sensible ways to extend the typespace for private use. I am hesitant to add a new `Key_Type` because we'd need to audit all the uses to make sure we don't accidentally break e.g. validation in the compiler. But given the `Key_Id` is limited to an `alnum_` pattern, couldn't we do something as simple as `T_*_MT_SHIFT_TO_CAPS`?

Then all I'd ask is that we add a comment to the keyman-touch-layout-file.ts explaining this private use.

Reply:
I think I should also add that this type is an input type for a data structure that's meant to be very rigid. We never want to see `undefined` or `S_foo` in .keyman_touch_layout files. So it wouldn't be appropriate to be extending the types for the input data structures, rather it would be appropriate for KeymanWeb to have its own 'subclassing' of the structures and types to avoid mixing implementation and data details.